code/game/machinery/firealarm.dm

```diff
@@ -35,12 +35,18 @@
 	var/area/my_area = null
 	///The current alarm state
 	var/alert_type = FIRE_CLEAR
+	///If it has a shelter inside
+	var/shelter = TRUE
```

Comment:
The shelter should be actually stored inside of the fire alarm instead of creating on use.